_source/_posts/2023-08-02-terraform-for-oidc-workshop.md

```diff
@@ -0,0 +1,230 @@
+
+```tf
+provider "okta" {
+  org_name    = "dev-1234567890"
+  base_url    = "okta.com"
+  client_id   = "00abc123FIXME00"
+  scopes      = ["okta.groups.manage", "okta.users.manage", "okta.policies.manage", "okta.apps.manage"]
+  private_key = "${path.module}/rsa.pem"
+}
+```
+
+
+Apply the Terraform, and you'll now see your "Todo app" in the applications list in the admin console! 
+
+### Get Secrets 
+
+The next step of the OIDC workshop involves using the `client_id` and `cilent_secret` from the newly created OIDC app. Terraform can retrieve these values for you or your automation to use elsewhere, using [outputs](https://developer.hashicorp.com/terraform/tutorials/configuration-language/outputs). 
+
+You could keep your outputs in `oidc_workshop.tf` for now, although maintaining all the outputs from your project in `outputs.tf` is also a recommended pattern. You can tell that the OAuth App resource exposes attributes named `client_id` and `client_secret` from the [Attributes Reference section](https://registry.terraform.io/providers/okta/okta/latest/docs/resources/app_oauth#attributes-reference) of its entry in the provider docs. 
+
+Note that the `client_secret` should be marked as a [sensitive value](https://developer.hashicorp.com/terraform/language/values/outputs#sensitive-suppressing-values-in-cli-output), so that it will be hidden by default and only shown when explicitly requested.
+
+```tf
+output "oidc_app_client_id" {
+  description = "Client ID for OIDC Workshop Todo app integration"
+  value       = okta_app_oauth.oidc_workshop.client_id
+}
+output "oidc_app_client_secret" {
+  description = "Client Secret for OIDC Workshop Todo app integration"
+  value       = okta_app_oauth.oidc_workshop.client_secret
+  sensitive   = true
+}
+```
+
+After adding these outputs to `oidc_workshop.tf`, run `terraform apply` to update Terraform's state with the output values. Then you can run `terraform output` to retrieve the all output values, or query an individual value with a command like `terraform output oidc_app_client_secret`. Sensitive values are only displayed when queried individually. 
+
+### Get Endpoint URLs
+
+Another manual task in the OIDC Workshop is finding various links that correspond to attributes of the Okta organization's OIDC server. Terraform can retrieve these values, as well! 
+
+
+# TODO FIXME everything below this point is wild guesswork and contains some deep misunderstandings about which IdP is "the" IdP in the context of the blog
+
+#### issuer
+[docs](https://registry.terraform.io/providers/okta/okta/latest/docs/data-sources/auth_server)
+
+```tf
+data "okta_auth_server" "default" {
+  name = "default"
+}
+output "issuer" {
+  description = "Issuer URL for org's default auth server"
+  value       = data.okta_auth_server.default.issuer
+}
+```
+
+#### Metadata URI
+[docs](https://registry.terraform.io/providers/okta/okta/latest/docs/data-sources/idp_oidc)
+```tf
+data "okta_idp_oidc" "default" {
```

Comment:
this may not be the correct TF data source to use. I feel like it should be https://registry.terraform.io/providers/okta/okta/latest/docs/data-sources/idp_oidc because there doesn't seem to be anything more appropriate in the data sources list, and it offers all the outputs I want from it, but I've never actually seen something like this succeed so my guess about its applicability could be the problem.